Pod/Classes/Deselection/UIViewController+Deselection.swift

```diff
@@ -30,64 +30,58 @@
 
 import UIKit
 
+public protocol SmoothlyDeselectableItems {
+    var indexPathsForSelectedItems: [IndexPath]? { get }
+    func selectItem(at indexPath: IndexPath?, animated: Bool)
+    func deselectItem(at indexPath: IndexPath, animated: Bool)
+}
+
+extension UITableView: SmoothlyDeselectableItems {
+    public var indexPathsForSelectedItems: [IndexPath]? { return indexPathsForSelectedRows }
+
+    public func selectItem(at indexPath: IndexPath?, animated: Bool) {
+        selectRow(at: indexPath, animated: animated, scrollPosition: .none)
+    }
+
+    public func deselectItem(at indexPath: IndexPath, animated: Bool) {
+        deselectRow(at: indexPath, animated: animated)
+    }
+}
+
+extension UICollectionView: SmoothlyDeselectableItems {
+    public func selectItem(at indexPath: IndexPath?, animated: Bool) {
+        selectItem(at: indexPath, animated: animated, scrollPosition: UICollectionViewScrollPosition())
+    }
+}
+
 public extension UIViewController {
 
     ///  Smoothly deselect selected rows in a table view during an animated
     ///  transition, and intelligently reselect those rows if the interactive
     ///  transition is canceled. Call this method from inside your view
     ///  controller's `viewWillAppear(_:)` method.
     ///
-    ///  - parameter tableView: The table view in which to perform deselection/reselection.
-    func rz_smoothlyDeselectRows(tableView tableView: UITableView?) {
-        let selectedIndexPaths = tableView?.indexPathsForSelectedRows ?? []
-
-        if let coordinator = transitionCoordinator() {
-            coordinator.animateAlongsideTransitionInView(parentViewController?.view, animation: { context in
-                selectedIndexPaths.forEach {
-                    tableView?.deselectRowAtIndexPath($0, animated: context.isAnimated())
-                }
-                }, completion: { context in
-                    if context.isCancelled() {
-                        selectedIndexPaths.forEach {
-                            tableView?.selectRowAtIndexPath($0, animated: false, scrollPosition: .None)
-                        }
-                    }
-            })
-        }
-        else {
-            selectedIndexPaths.forEach {
-                tableView?.deselectRowAtIndexPath($0, animated: false)
-            }
-        }
-    }
-
-    ///  Smoothly deselect selected items in a collection view during an animated
-    /// transition, and intelligently reselect those items if the interactive
-    /// transition is canceled. Call this method from inside your view
-    ///  controller's `viewWillAppear(_:)` method.
-    ///
-    ///  - parameter collectionView: The table view in which to perform deselection/reselection.
-    func rz_smoothlyDeselectItems(collectionView collectionView: UICollectionView?) {
-        let selectedIndexPaths = collectionView?.indexPathsForSelectedItems() ?? []
+    ///  - parameter tableView: The de/selectable view in which to perform deselection/reselection.
```

Comment:
#nth `(de)selectable`, except #req it looks like this method doesn't have this parameter any more. Are we linting for mismatched parameters? It's `valid_docs` in Swiftlint.